# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: arm_1
  joint_model_group_pose: todo_no_pose_selected
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - arm1_joint1
    - arm1_joint2
    - arm1_joint3
    - arm1_joint4
    - arm1_joint5
    - arm2_joint1
    - arm2_joint2
    - arm2_joint3
    - arm2_joint4
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
controller_list:
  []
arm1_position_controller:
  type: position_controllers/JointPositionController
  joints:
    - arm1_joint1
    - arm1_joint2
    - arm1_joint3
    - arm1_joint4
    - arm1_joint5
  gains:
    arm1_joint1:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm1_joint2:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm1_joint3:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm1_joint4:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm1_joint5:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
arm2_position_controller:
  type: position_controllers/JointPositionController
  joints:
    - arm2_joint1
    - arm2_joint2
    - arm2_joint3
    - arm2_joint4
  gains:
    arm2_joint1:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm2_joint2:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm2_joint3:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
    arm2_joint4:
      p: 100
      d: 1
      i: 1
      i_clamp: 1
